ricecooker/sushi_bar_client.py

```diff
@@ -178,59 +241,51 @@ def report_statistics(self, files):
 
-    def __init__(self, url):
-        threading.Thread.__init__(self)
-        self.stop_event = threading.Event()
-        self.url = url
-        self.ws = None
-        self.connect_lock = threading.Lock()
 
-    def __connect(self):
-        self.ws = websocket.WebSocketApp(
-            self.url,
-            on_message=self.on_message
-        )
 
-    def run(self):
-        """
-        If the connection drops, then run_forever will terminate and a
-        reconnection attempt will be made.
-        """
-        while True:
-            self.connect_lock.acquire()
-            if self.stopped():
-                return
-            self.__connect()
-            self.connect_lock.release()
-            self.ws.run_forever()
+
+# CONTROL
+################################################################################
+
+class ControlWebSocket(ReconnectingWebSocket):
+    def __init__(self, chef, args, options):
+        self.chef = chef
+        self.cliargs = args
+        self.clioptions = options
+        self.channel = chef.get_channel()
+        if not self.channel:
+            raise SushiBarNotSupportedException(
+                'Chef does not implement create_channel')
+        self.thread = None
+        print('Channel id %s' % self.channel.get_node_id().hex)
+        url = config.sushi_bar_control_url(self.channel.get_node_id().hex)
+        ReconnectingWebSocket.__init__(self, url)
 
     def on_message(self, ws, message):
-        pass
+        message = json.loads(message)
 
-    def send(self, data):
-        """
-        This method keeps trying to send a message relying on the run method
-        to reopen the websocket in case it was closed.
-        """
-        while not self.stopped():
-            try:
-                self.ws.send(data)
-                return
-            except websocket.WebSocketConnectionClosedException:
-                print('WebSocket closed, retrying send.')
-                time.sleep(0.1)
+        # args and options from SushiBar overrride command line args and options
+        args = self.cliargs
+        options = self.clioptions
+        if 'args' in message:
+            args.update(message['args'])
+        if 'options' in message:
+            options.update(message['options'])
+
+        if message['command'] == 'start':
+            if not self.thread or not self.thread.isAlive():
+                self.thread = threading.Thread(
+                    target=self.chef.run,
+                    args=(args, options),
+                )
+                self.thread.start()
+            else:
+                print('Already running')
+        else:
+            print('Command not supported: %s' % message['command'])
 
-    def stop(self):
-        self.connect_lock.acquire()
-        self.stop_event.set()
-        self.ws.close()
-        self.connect_lock.release()
 
-    def stopped(self):
-        return self.stop_event.is_set()
 
 
 class LoggingHandler(logging.Handler):
```

Comment:
what do you think about moving this class to the monitoring section of the file?